core/trino-main/src/main/java/io/trino/memory/LowMemoryKiller.java

```diff
@@ -61,8 +94,20 @@ public String toString()
             return toStringHelper(this)
                     .add("queryId", queryId)
                     .add("memoryReservation", memoryReservation)
+                    .add("queryStats", queryStats)
+                    .add("taskStats", taskInfos)
                     .add("retryPolicy", retryPolicy)
                     .toString();
         }
     }
+
+    @Retention(RUNTIME)
+    @Target({FIELD, PARAMETER, METHOD})
+    @Qualifier
+    @interface ForQuery {}
```

Comment:
nit: Maybe `ForTaskLowMemoryKiller` / `ForQueryLowMemoryKiller`?

Reply:
Yeah - turned out not needed - will remove